src/database/testcases.tsx

```diff
@@ -0,0 +1,196 @@
+import { deleteFacility, saveFacility } from "./index";
```

Comment:
Note:  This file isn't meant to be merged in along with this PR, but it is probably the most important thing to review.  The code above is all relatively straightforward (I hope 🤞).  This is where I spent most of my time with this PR, creating test cases to give us confidence that things work as expected.  Please review the scenarios I tested and feel free to comment with more and I'll try them out locally.

Reply:
This is fantastic! I wouldn't delete this, actually. Even though it's not structured to run as unit tests, I would still move it into a `src/database/__tests__` directory and maybe add a comment at the top that this is a file for operating manual tests.